server/impl/src/main/java/com/walmartlabs/concord/server/user/UserResource.java

```diff
@@ -79,6 +79,10 @@ public CreateUserResponse createOrUpdate(@ApiParam @Valid CreateUserRequest req)
 
         UUID id = userManager.getId(username, req.getUserDomain(), type).orElse(null);
         if (id == null) {
+            if (!userManager.validateInfo(username, req.getUserDomain(), type)) {
+                throw new ConcordApplicationException("Validating userinfo failed for user id: "+id+" of type: " + type, Status.BAD_REQUEST);
```

Comment:
This is not actionable, it's not going to return anything that would tell the user why the validation failed. Perhaps it should call `UserManager#getOrCreate` instead - assuming the validation there provides the actual error like [the previous implementation did](https://github.com/walmartlabs/concord/blob/master/server/impl/src/main/java/com/walmartlabs/concord/server/security/ldap/LdapUserInfoProvider.java#L84).

Reply:
implemented as commented